src/main/java/org/openrewrite/java/testing/cleanup/RemoveVisibleForTestingAnnotationWhenUsedInProduction.java

```diff
@@ -0,0 +1,196 @@
+                "This recipe should not be used in an environment where QA tooling acts on the `@VisibleForTesting` annotation.";
+    }
+
+    @Override
+    public Scanned getInitialValue(ExecutionContext ctx) {
+        return new Scanned();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Scanned acc) {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.FieldAccess visitFieldAccess(J.FieldAccess fieldAccess, ExecutionContext ctx) {
+                // Mark classes
+                if (fieldAccess.getTarget().getType() instanceof JavaType.Class) {
+                    JavaType.Class type = (JavaType.Class) fieldAccess.getTarget().getType();
+                    if (!acc.classes.contains(type)) {
+                        type.getAnnotations().forEach(annotation -> {
+                            if ("VisibleForTesting".equals(annotation.getClassName())) {
+                                J.CompilationUnit compilationUnit = getCursor().firstEnclosing(J.CompilationUnit.class);
+                                if (compilationUnit != null && compilationUnit.getSourcePath().startsWith("src/main")) {
+                                    acc.classes.add(type);
+                                }
+                            }
+                        });
+                    }
+                }
+                // Mark fields
+                if(fieldAccess.getName().getFieldType() != null && !acc.fields.contains(fieldAccess.getName().getFieldType())) {
+                    fieldAccess.getName().getFieldType().getAnnotations().forEach(annotation -> {
+                        if ("VisibleForTesting".equals(annotation.getClassName())) {
+                            J.CompilationUnit compilationUnit = getCursor().firstEnclosing(J.CompilationUnit.class);
+                            if (compilationUnit != null && compilationUnit.getSourcePath().startsWith("src/main")) {
+                                acc.fields.add(fieldAccess.getName().getFieldType());
+                            }
+                        }
+                    });
+                }
+
+                return super.visitFieldAccess(fieldAccess, ctx);
+            }
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                if (method.getMethodType() != null) {
+                    if (!acc.methods.contains(method.getMethodType())) {
+                        method.getMethodType().getAnnotations().forEach(annotation -> {
+                            if ("VisibleForTesting".equals(annotation.getClassName())) {
+                                J.CompilationUnit compilationUnit = getCursor().firstEnclosing(J.CompilationUnit.class);
+                                if (compilationUnit != null && compilationUnit.getSourcePath().startsWith("src/main")) {
```

Comment:
This should check the `JavaSourceSet` marker to see if it has name `main`. Not every project will use the Maven idiom for source set pathing.